security-analysis/security-analysis-api/src/main/java/com/powsybl/security/limitreduction/AbstractContingencyWiseReducedLimitsComputer.java

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.security.limitreduction;
+
+import com.powsybl.contingency.ContingencyContext;
+import com.powsybl.iidm.criteria.NetworkElementCriterion;
+import com.powsybl.iidm.criteria.NetworkElementVisitor;
+import com.powsybl.iidm.criteria.duration.AbstractTemporaryDurationCriterion;
+import com.powsybl.iidm.criteria.duration.LimitDurationCriterion;
+import com.powsybl.iidm.criteria.translation.NetworkElement;
+import com.powsybl.iidm.network.LimitType;
+import com.powsybl.iidm.network.ThreeSides;
+import com.powsybl.iidm.network.limitreduction.computer.AbstractReducedLimitsComputer;
+import com.powsybl.iidm.network.limitreduction.computer.ReducedLimitsComputer;
+import com.powsybl.iidm.network.limitreduction.result.LimitsContainer;
+import com.powsybl.iidm.network.limitreduction.result.UnalteredLimitsContainer;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.powsybl.contingency.ContingencyContextType.*;
+
+/**
+ * Abstract class responsible for computing reduced limits using a {@link LimitReductionList}.
+ *
+ * @author Sophie Frasnedo {@literal <sophie.frasnedo at rte-france.com>}
+ * @author Olivier Perrin {@literal <olivier.perrin at rte-france.com>}
+ */
+public abstract class AbstractContingencyWiseReducedLimitsComputer<P, L> extends AbstractReducedLimitsComputer<P, L> {
```

Comment:
AbstractContingencyWiseReducedLimitsComputer? What does it mean?

Reply:
Seen with @flo-dup: this class was renamed as `AbstractLimitReductionsApplier`.